main/http_server/http_server.c

```diff
@@ -831,11 +900,87 @@ static esp_err_t GET_system_statistics(httpd_req_t * req)
         return ESP_OK;
     }
 
+    char * buf = NULL;
+    size_t bufLen = httpd_req_get_url_query_len(req) + 1;
+    bool dataSelection[SRC_NONE] = {false};
+    bool selectionCheck = false;
+    int prebuffer = 0;
+
+    // Check query parameters
+    if (1 < bufLen) {
+        buf = (char *)malloc(bufLen);
+        if (buf) {
+            if (httpd_req_get_url_query_str(req, buf, bufLen) == ESP_OK) {
+                char * param = strtok(buf, "&");
```

Comment:
You can do `httpd_uri_decode` here, and get rid of the `bool urlEnc` in `strToDataSource` and the double string constants at the top.